Application/EdFi.Ods.Api/ExceptionHandling/ModelStateKeyConverter.cs

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Autofac.Extras.DynamicProxy;
+using EdFi.Ods.Common.Models.Resource;
+
+namespace EdFi.Ods.Api.ExceptionHandling;
+
+[Intercept("cache-model-state-key")]
```

Comment:
An interceptor has been added here to prevent the same path from ever being translated twice. This may or may not be a good optimization, but the idea is to reduce the number of string allocations that would be caused by repeatedly translating the same model paths to JSON Paths.